go/consensus/tendermint/db/badger/badger.go

```diff
@@ -474,11 +474,24 @@ func (ba *badgerDBBatch) WriteSync() error {
 
 	for _, cmd := range ba.cmds {
 		if err := cmd.Execute(tx); err != nil {
-			ba.db.logger.Error("failed to execute command in Tx",
-				"err", err,
-				"cmd", cmd,
-			)
-			return err
+			if err == badger.ErrTxnTooBig {
```

Comment:
Since this looks a bit unconventional, maybe add a similar comment as the one in the issue description here in code as well. Together with a reference to the badger docs.

Reply:
Done :)

Reply:
Oh, but does this mean that we can end up with some "commands" being committed and some not? Since, this actually commits the old transaction and starts a new one.

Is this ok with the expected semantics of `WriteSync`?

Reply:
Good point.  There's some discussion on Badger's issue tracker here: https://github.com/dgraph-io/badger/issues/1242

It doesn't look like there's any other way to implement this.

So far, the only time I've encountered this error is when running Tendermint's 0.34->0.35 migration scripts, which consistently fail here because of `ErrTxnTooBig`.  It seems all normal transactions are small enough to go through with sync semantics unchanged, but really large ones can indeed commit half-way.  Given that we've never encountered this before, it looks like the only place where really large transactions are currently used is in the migration scripts and those are written in a way that handles partially-migrated databases, so it should be fine.

If we want to be extra safe, I can close this PR and change the Tendermint migration scripts to use `WriteBatch` instead, which does the same thing internally, except it doesn't have atomic semantics.

@kostko What do you think?

Reply:
If we're confident that we'll never see this issue apart from in the migration scripts, that's fine as well.

Reply:
Yeah I think it is ok. Would be safer if there was an explicit flag (only enabled during migrations), but seeing how the DB config has this Tendermint's interface it would be rather clunky.